### PR TITLE
-

### PR DESCRIPTION
**Pull request by guffy1234**

### CODE REVIEW

File: src/HttpRouter.h [hunk 1]
```diff
@@ -257,27 +221,21 @@ class HttpRouter {
...
-
-        /* We have that method on record, let's iterate it */
-        if (parent != &tree) {
-            return executeHandlers(parent, 0, userData);
-        } else {
-            /* We did not find any handler for this method.
-             * You may want to re-route with "*" as method. */
-            return false;
-        }
+		for (auto &p : tree.children) {
```
Comment (alexhultman) [inline src/HttpRouter.h]: This is basically

```
for (auto &c : tree->children) {
	if (c->name == method) {
		return executeHandlers(c.get(), 0, userData);
	}
}
return false;
```

**Comment by guffy1234** [comment]
Yes. Have no idea why github shows changes this way.
This is the file
https://github.com/guffy1234/uWebSockets/blob/master/src/HttpRouter.h

File: src/HttpRouter.h (hunk 1 above)
Comment (guffy1234) [reply src/HttpRouter.h]: Yes. Have no idea why github shows changes this way.
This is the file
https://github.com/guffy1234/uWebSockets/blob/master/src/HttpRouter.h

**Comment by alexhultman** [comment]
Hello,

I can look at this and see what's easy to put in. The router is still WIP and should be rewritten completely when it's behavior is known.

**Comment by guffy1234** [comment]
Ok. Honestly it looks like designed to support high amount of routes. You trying to save space in the nodes with indirect handlers. May be I'm not too familiar with web applications, but imho probably it's unusual case. And tree-like routing makes difficult to introduce regex matches, especially multi segment patterns

**Comment by alexhultman** [comment]
It doesn't do regex. Regex are slow very slow

**Comment by guffy1234** [comment]
I have changed my opinion. Tree based routing is enough. Whenever i will need something custom i can do it as second layer routing in a handler. Like filesystem streamer does.

**Pull request closed without merging**